packages/assets-controllers/src/TokenBalancesController.ts

```diff
@@ -1,141 +1,227 @@
-      interval: 180000,
-      tokens: [],
-    };
-    this.defaultState = { contractBalances: {} };
-    this.initialize();
-    onTokensStateChange(({ tokens, detectedTokens }) => {
-      this.configure({ tokens: [...tokens, ...detectedTokens] });
-      this.updateBalances();
+  constructor({
+    interval = DEFAULT_INTERVAL,
+    tokens = [],
+    disabled = false,
+    onTokensStateChange,
+    getSelectedAddress,
+    getERC20BalanceOf,
+    messenger,
+    state = {},
+  }: TokenBalancesControllerOptions) {
+    super({
+      name: controllerName,
+      metadata,
+      messenger,
+      state: {
+        ...getDefaultState(),
+        ...state,
+      },
     });
-    this.getSelectedAddress = getSelectedAddress;
-    this.getERC20BalanceOf = getERC20BalanceOf;
+
+    this.#disabled = disabled;
+    this.#interval = interval;
+    this.#tokens = tokens;
+
+    onTokensStateChange(this.#tokensStateChangeListener.bind(this));
+
+    this.#getSelectedAddress = getSelectedAddress;
+    this.#getERC20BalanceOf = getERC20BalanceOf;
+
     this.poll();
   }
 
+  /*
+   * Tokens state changes listener.
+   */
+  #tokensStateChangeListener({ tokens, detectedTokens }: TokensState) {
+    this.#tokens = [...tokens, ...detectedTokens];
+    this.updateBalances();
+  }
+
+  /**
+   * Allows controller to update tracked tokens contract balances.
+   */
+  enable() {
+    this.#disabled = false;
+  }
+
+  /**
+   * Blocks controller from updating tracked tokens contract balances.
+   */
+  disable() {
+    this.#disabled = true;
+  }
+
   /**
    * Starts a new polling interval.
    *
    * @param interval - Polling interval used to fetch new token balances.
    */
   async poll(interval?: number): Promise<void> {
-    interval && this.configure({ interval }, false, false);
-    this.handle && clearTimeout(this.handle);
+    if (interval) {
+      this.#interval = interval;
+    }
+
+    if (this.#handle) {
+      clearTimeout(this.#handle);
+    }
+
     await safelyExecute(() => this.updateBalances());
-    this.handle = setTimeout(() => {
-      this.poll(this.config.interval);
-    }, this.config.interval);
+
+    this.#handle = setTimeout(() => {
+      this.poll(this.#interval);
+    }, this.#interval);
   }
 
   /**
    * Updates balances for all tokens.
    */
   async updateBalances() {
-    if (this.disabled) {
+    if (this.#disabled) {
       return;
     }
-    const { tokens } = this.config;
-    const newContractBalances: { [address: string]: BN } = {};
-    for (const token of tokens) {
+
+    const newContractBalances: ContractBalances = {};
+    for (const token of this.#tokens) {
       const { address } = token;
       try {
-        newContractBalances[address] = await this.getERC20BalanceOf(
-          address,
-          this.getSelectedAddress(),
-        );
+        newContractBalances[address] = (
+          await this.#getERC20BalanceOf(address, this.#getSelectedAddress())
+        ).toString(16);
```

Comment:
Following my comment about this, if we decide to prefix hex values with `0x`, then this would be:

``` suggestion
        newContractBalances[address] = toHex(
          await this.#getERC20BalanceOf(address, this.#getSelectedAddress())
        );